src/access/wac.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { internal_fetchAcl } from "../acl/acl.internal";
+import { getAgentAccess as getAgentAccessWac } from "../acl/agent";
+import { WebId, WithServerResourceInfo } from "../interfaces";
+import { internal_defaultFetchOptions } from "../resource/resource";
+import { Access } from "./universal";
+import { Access as AclAccess } from "../acl/acl";
+
+// Setting WacAccess = Required<Access> enforces that any change on Access will
+// reflect on WacAccess, but WacAccess has additional restrictions.
+type WacAccess = Required<Access> &
```

Comment:
`Required<>` is the opposite of `Partial<>`, i.e. it makes all properties _not_ be `undefined`. Which made sense when I thought they had to be `true` or `false` for WacAccess, but if they're `true` or `undefined`, less so :)

(In fact, I'm wondering whether the union type with Access is actually needed/might even override the other type (in the sense that setting a property of a `WacAccess` to `false` would still be allowed). I think as long as our universal API, which has just `Access` as a return type`, directly passes on the return value of the WAC-specific API, we're still safe in terms of getting a warning when either of those types changes.)

Reply:
Removed the union with Access: 90ec90d4c1de96c4ef16c609e6047a7c951a8cb6